File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
       unordered_map <int, int> hash;
       int n = nums.size();
       for (int i = 0; i < n; i++) {
        int diff = target - nums[i];
        if (hash.find(diff) != hash.end()) {
            return {hash[diff], i};
        }
        hash.insert({nums[i], i});
       }
       return {};
    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        n = len(numbers)
        l = 0;
        r = n - 1
        while l < r:
            if numbers[l] + numbers[r] > target:
                r -= 1
            elif numbers[l] + numbers[r] < target:
                l += 1
            else:
                return [l+1, r+1]


